nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/directory/CreateVocabularyEntry.java

```diff
@@ -0,0 +1,112 @@
+import org.nuxeo.ecm.core.schema.SchemaManager;
+import org.nuxeo.ecm.core.schema.types.Schema;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.api.DirectoryService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Adds a new entry ion a vocabulary.
+ * <p>
+ * Notice: This is for a nuxeo Vocabulary, which is a specific kind of Directory. This code expects the following:
+ * <ul>
+ * <li>The vocabulary schema <i>must</i> have <code>id</code>, <code>label</code>, <code>obsolete</code> and
+ * <code>ordering</code> fields</li>
+ * <li>If it is hierarchical, it must also have the <code>parent</code> field</li>
+ * </ul>
+ */
+@Operation(id = CreateVocabularyEntry.ID, category = Constants.CAT_SERVICES, label = "Vocabulary: Add Entry", description = "Add a new entry in the <i>name</i> directory only if <i>id</i> is not found (an existing entry is not updated). If <i>label</i> is empty, it is set to the id. WARNING: Current user must have enough rights to write in a directory.")
+public class CreateVocabularyEntry {
+
+    public static final String ID = "Directory.CreateVocabularyEntry";
+
+    // Caching infos about vocabularies, to avoid getting the schema and testing the "parent" field at every call.
+    // This is because passing a Map with "parent" to a simple vocabulary (non hierarchical) throws an error
+    protected static HashMap<String, Boolean> vocabularyAndHasParent = new HashMap<>();
+
+    @Context
+    protected DirectoryService directoryService;
+
+    @Context
+    protected SchemaManager schemaManager;
+
+    @Param(name = "name")
+    protected String name;
+
+    @Param(name = "id")
+    protected String id;
+
+    @Param(name = "label", required = false)
+    protected String label;
+
+    @Param(name = "parent", required = false)
+    protected String parent = "";
+
+    @Param(name = "obsolete", required = false)
+    protected long obsolete = 0;
+
+    @Param(name = "ordering", required = false)
+    protected long ordering = 0;
+
+    @OperationMethod
+    public void run() {
+
+        if (StringUtils.isNotBlank(id)) {
+            boolean hasParent;
+            if (vocabularyAndHasParent.get(name) == null) {
+                String dirSchema = directoryService.getDirectorySchema(name);
+                Schema schema = schemaManager.getSchema(dirSchema);
+                hasParent = schema.hasField("parent");
+                vocabularyAndHasParent.put(name, hasParent);
+            } else {
+                hasParent = vocabularyAndHasParent.get(name);
+            }
+
+            Session directorySession = directoryService.open(name);
+            if (!directorySession.hasEntry(id)) {
+                Map<String, Object> entry = new HashMap<>();
+                entry.put("id", id);
+                if (StringUtils.isEmpty(label)) {
+                    label = id;
+                }
+                entry.put("label", label);
```

Comment:
Remove the previous `if` and use `entry.put("label", defaultIfEmpty(label, id)))`